### PR TITLE
Org And Name Area Clickable

### PR DESCRIPTION
**Pull request by jayfyle**
## Menu Click To Profile Fix
App_menu

Now clicking on the name of the user on top of the menu will take the user to the profile page.

### Edit 1: 

https://user-images.githubusercontent.com/115472256/203004499-2bdb5cf1-eb2a-4d52-830d-992db2adb818.mov
Ripple effect added and click handler now relies on event emitter from sidemenu-header to parent component. 

### Edit 2: 

https://user-images.githubusercontent.com/115472256/203482899-c757ec2f-be32-41a2-8e30-c22ad1518fb4.mov

Ripple effect improved and refactor

### CODE REVIEW

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Add ripple effect for the click so that the user understands that some action is happening on the click and attach a recording of the same

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts [hunk 1]
```diff
@@ -12,7 +14,12 @@ export class SidemenuHeaderComponent implements OnInit {
 
   @Input() activeOrg: Org;
 
-  constructor() {}
+  constructor(private router: Router, private menuController: MenuController) {}
+
+  onClickProfile() {
```
Comment (yksolanki9) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts:19]: Redirection should happen from the parent component, emit an event on click and Output it to the parent component

Comment (yksolanki9) [reply src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts:19]: This is a presentational component - https://blog.angular-university.io/angular-2-smart-components-vs-presentation-components-whats-the-difference-when-to-use-each-and-why/

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Add ripple effect as discussed

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.html [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-              class="sidemenu-header__content-container__icon"
-              src="../../../../../assets/svg/organization.svg"
-              slot="icon-only"
-            ></ion-icon>
-            <div class="sidemenu-header__content-container__sub-title">{{ activeOrg?.name }}</div>
-          </div>
-        </ion-col>
-      </ion-row>
-    </ion-grid>
+    <div matRipple [matRippleColor]="'rgba(255, 255, 255, 0.3)'" (click)="onClickProfile($event)">
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.html:3]: why do we need the matRippleColor here? is this by design?

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts [hunk 3]
```diff
@@ -12,7 +13,13 @@ export class SidemenuHeaderComponent implements OnInit {
 
   @Input() activeOrg: Org;
 
+  @Output() profileClick = new EventEmitter();
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts:16]: `profileClicked`?

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts [hunk 4]
```diff
@@ -12,7 +13,13 @@ export class SidemenuHeaderComponent implements OnInit {
...
+
   constructor() {}
 
+  onClickProfile(event: Event) {
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.ts:20]: `onProfileClick`?

File: src/app/shared/components/sidemenu/sidemenu.component.ts [hunk 5]
```diff
@@ -370,6 +374,13 @@ export class SidemenuComponent implements OnInit {
     ].filter((sidemenuItem) => sidemenuItem.isVisible);
   }
 
+  goToProfile(event) {
```
Comment (Dimple16) [inline src/app/shared/components/sidemenu/sidemenu.component.ts:377]: event type?

**Review by Dimple16 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — APPROVED** [review]

File: src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.html [hunk 6]
```diff
@@ -1,26 +1,28 @@
...
-              class="sidemenu-header__content-container__icon"
-              src="../../../../../assets/svg/organization.svg"
-              slot="icon-only"
-            ></ion-icon>
-            <div class="sidemenu-header__content-container__sub-title">{{ activeOrg?.name }}</div>
-          </div>
-        </ion-col>
-      </ion-row>
-    </ion-grid>
+    <div matRipple [matRippleColor]="'rgba(255, 255, 255, 0.3)'" (click)="onProfileClicked($event)">
```
Comment (yksolanki9) [inline src/app/shared/components/sidemenu/sidemenu-header/sidemenu-header.component.html:3]: `matRippleColor="rgba(255, 255, 255, 0.3)"`

File: src/app/shared/components/sidemenu/sidemenu.component.ts [hunk 7]
```diff
@@ -370,6 +374,13 @@ export class SidemenuComponent implements OnInit {
     ].filter((sidemenuItem) => sidemenuItem.isVisible);
   }
 
+  goToProfile(event: Event) {
+    if (event.isTrusted) {
```
Comment (yksolanki9) [inline src/app/shared/components/sidemenu/sidemenu.component.ts:378]: Can skip this check

File: src/app/shared/components/sidemenu/sidemenu.component.ts [hunk 8]
```diff
@@ -370,6 +374,13 @@ export class SidemenuComponent implements OnInit {
     ].filter((sidemenuItem) => sidemenuItem.isVisible);
   }
 
+  goToProfile(event: Event) {
```
Comment (yksolanki9) [inline src/app/shared/components/sidemenu/sidemenu.component.ts:377]: `goToProfilePage` - to make it more evident that we're navigating to another page

**Review by yksolanki9 — APPROVED** [review]
Left some minor comments

**Pull request merged**